Div component

MonPote opened the pull request:
**Component**:

<!-- E.g. 'Layout', 'Table', 'build', 'tests'... -->

**Description**:

**Design**:

**Breaking Changes**:

- [] Breaking Changes


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #ISSUE_NUMBER

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

MonPote committed:
spacedbox: first version of spacebox

bert-e commented:
# Hello monpote,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/core-ui/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @rachedbenmustapha

* @carlito-scality

* @monpote

* @jbwatenbergscality

* @nicolas2bert

JBWatenbergScality commented on src/lib/components/div/Div.js:
```diff
@@ -0,0 +1,112 @@
+const mr = createSpacingFn('mr', ['margin-right']);
+const mb = createSpacingFn('mb', ['margin-bottom']);
+const ml = createSpacingFn('ml', ['margin-left']);
+const mx = createSpacingFn('mx', ['margin-top', 'margin-bottom']);
+const my = createSpacingFn('my', ['margin-left', 'margin-right']);
```
I think x and y axis are reversed here. In the common spirit x should stand for the horizontal axis I believe. (Tailwind also implemented the x axis against left/right properties : https://tailwindcss.com/docs/padding#add-horizontal-padding)

MonPote replied:
Indeed that was a mistake

JBWatenbergScality reviewed (commented):
Additionally to the raw component we might want to document/show some example of the composability of this approach when it comes to use those spacing utils with other components

MonPote committed:
spacedbox: add spacedbox stories

MonPote committed:
remove noisy ci test

Currently `npm run test` is only snapshot test.
It has the benefit to force the developer to double check components
that might be affected by some style change and avoid regresion.
However, the current behavior generate more noise by updating almost
all the time all the snapshot.
This does not help developer to find bug, we will need to find a solution
to properly catch regresion.

JBWatenbergScality commented on stories/div.stories.js:
```diff
@@ -0,0 +1,136 @@
+      <Div as={BoldP} my={14} p={14}>
+        Bold {'"</p>"'} with my 14 and p 14
+      </Div>
+      <Div
+        as={DisplayTextComp}
+        text={'Div + React component + nested styled component'}
+        m={12}
+        p={14}
+      ></Div>
```
This is great, I'm wondering how does it affect the typing ? Do we get props autocompletion/validation for both Div and  DisplayTextComp component ?

MonPote commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`




Please check the status of the associated issue None.

Goodbye monpote.
